### PR TITLE
Implemented changes for custom python test

### PR DESCRIPTION
**Pull request by rquidute**
### What changed
Updated the files related to custom python tests in order to the be in accordance with the changes with the way to retrieve test steps names.

### Related Issue
https://github.com/project-chip/certification-tool/issues/484 

### Testing
Unit test updated and passing
Screenshot 2024-11-12 at 20 21 53

TH UI List the custom test case
Screenshot 2024-11-12 at 20 22 27

TH UI execution runs successfully the Custom Python test case.
Screenshot 2024-11-12 at 20 22 50

### CODE REVIEW

**Review by hiltonlima — APPROVED** [review]
Just check the CI errors

File: app/tests/conftest.py [hunk 1]
```diff
@@ -27,12 +32,18 @@
 from sqlalchemy import create_engine
 from sqlalchemy.orm import Session, sessionmaker
 
+# isort: split - DO NOT MOVE THIS IMPORT, this should be before any app.*
+from .sdk_container_mock import mock_instance  # Mock is already setup when this imports
```
Comment (antonio-amjr) [inline app/tests/conftest.py:36]: Where is this being used?

Comment (rquidute) [reply app/tests/conftest.py:36]: The initial goal was to initialize `mock_instance` before any app.*. 
I moved this import to the `__init__.py`, and it worked same way. Thanks, good catch!

**Comment by rquidute** [comment]
spelling and unit test are still failing. At first, it does not make sense to me. I will merge this PR now and check this afterwards. In case, I will open a new PR to fix this CI issues.

File: cspell.json [hunk 2]
```diff
@@ -114,7 +114,8 @@
         "test_collections/matter/sdk_tests/sdk_checkout",
         "test_collections/matter/python_tests/docs/common_test_failures/",
         "sdk_patch",
-        ".devcontainer"
+        ".devcontainer",
+        "test_collections/matter/sdk_tests/*.json"
```
Comment (ccruzagralopes) [inline cspell.json:118]: Should the json file for non-custom test cases also be ignored?

File: test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py [hunk 3]
```diff
@@ -170,26 +177,32 @@ def sdk_mandatory_python_test_collection(
 
 def custom_python_test_collection(
     python_test_folder: SDKTestFolder = CUSTOM_PYTHON_TEST_FOLDER,
+    tests_file_path: Path = CUSTOM_PYTHON_TESTS_PARSED_FILE,
 ) -> Optional[PythonCollectionDeclaration]:
+    asyncio.run(
+        generate_python_test_json_file(
+            test_folder=CUSTOM_PYTHON_SCRIPTS_FOLDER,
+            json_output_file=COMPLETE_JSON_OUTPUT_FILE_FOLDER
+            / "custom_python_tests_info.json",
```
Comment (ccruzagralopes) [inline test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py:186]: Please correct me if I'm wrong, but it seems like `COMPLETE_JSON_OUTPUT_FILE_FOLDER` is the same as `SDK_TESTS_ROOT`.
In that case, `COMPLETE_JSON_OUTPUT_FILE_FOLDER / "custom_python_tests_info.json"` is the same as `CUSTOM_PYTHON_TESTS_PARSED_FILE`.
If that's true, then I suggest using `CUSTOM_PYTHON_TESTS_PARSED_FILE` directly.

Comment (ccruzagralopes) [reply test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py:186]: Also, I don't understand why a constant that starts with "COMPLETE" actually always needs to be used with a complement. Maybe renaming it to just `JSON_OUTPUT_FILE_FOLDER` would be better?

File: test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py [hunk 4]
```diff
@@ -170,26 +177,32 @@ def sdk_mandatory_python_test_collection(
 
 def custom_python_test_collection(
     python_test_folder: SDKTestFolder = CUSTOM_PYTHON_TEST_FOLDER,
+    tests_file_path: Path = CUSTOM_PYTHON_TESTS_PARSED_FILE,
 ) -> Optional[PythonCollectionDeclaration]:
+    asyncio.run(
+        generate_python_test_json_file(
+            test_folder=CUSTOM_PYTHON_SCRIPTS_FOLDER,
```
Comment (ccruzagralopes) [inline test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py:184]: Are `SDK_PYTHON_TEST_FOLDER` and `CUSTOM_PYTHON_TEST_FOLDER` (declared in this file) still being used? If not, please delete them.
Maybe I misunderstood, but I think these are the old folders and now both folder are defined in `list_python_tests_classes` and we don't need these anymore.

**Review by ccruzagralopes — APPROVED** [review]

File: cspell.json (hunk 2 above)
Comment (rquidute) [reply cspell.json:118]: The custom json file is indicated at .ignore file, this is already the non-custom

Comment (ccruzagralopes) [reply cspell.json:118]: I got confused with the .gitignore file, sorry!

File: test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py (hunk 3 above)
Comment (rquidute) [reply test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py:186]: Good catch!

**Review by antonio-amjr — APPROVED** [review]

**Review by ccruzagralopes — APPROVED** [review]

**Pull request merged**